yt_dlp/extractor/gedidigital.py

```diff
@@ -1,198 +1,104 @@
-            ['twitter:title', 'og:title'], webpage, fatal=True)
-        player_data = re.findall(
-            r"PlayerFactory\.setParam\('(?P<type>format|param)',\s*'(?P<name>[^']+)',\s*'(?P<val>[^']+)'\);",
-            webpage)
-
+        data = self._search_json(
+            r'BrightcoveVideoPlayerOptions\s*=', webpage, 'Brightcove video player options', video_id, transform_source=js_to_json,
+        )
+        streams = json.loads(data['videoSrc'])
         formats = []
-        duration = thumb = None
-        for t, n, v in player_data:
-            if t == 'format':
-                if n in ('video-hds-vod-ec', 'video-hls-vod-ec', 'video-viralize', 'video-youtube-pfp'):
-                    continue
-                elif n.endswith('-vod-ak'):
-                    formats.extend(self._extract_akamai_formats(
-                        v, video_id, {'http': 'media.gedidigital.it'}))
-                else:
-                    ext = determine_ext(v)
-                    if ext == 'm3u8':
-                        formats.extend(self._extract_m3u8_formats(
-                            v, video_id, 'mp4', 'm3u8_native', m3u8_id=n, fatal=False))
-                        continue
-                    f = {
-                        'format_id': n,
-                        'url': v,
-                    }
-                    if ext == 'mp3':
-                        abr = int_or_none(self._search_regex(
-                            r'-mp3-audio-(\d+)', v, 'abr', default=None))
-                        f.update({
-                            'abr': abr,
-                            'tbr': abr,
-                            'acodec': ext,
-                            'vcodec': 'none',
-                        })
-                    else:
-                        mobj = re.match(r'^video-rrtv-(\d+)(?:-(\d+))?$', n)
-                        if mobj:
-                            f.update({
-                                'height': int(mobj.group(1)),
-                                'vbr': int_or_none(mobj.group(2)),
-                            })
-                        if not f.get('vbr'):
-                            f['vbr'] = int_or_none(self._search_regex(
-                                r'-video-rrtv-(\d+)', v, 'abr', default=None))
-                    formats.append(f)
-            elif t == 'param':
-                if n in ['image_full', 'image']:
-                    thumb = v
-                elif n == 'videoDuration':
-                    duration = int_or_none(v)
-
-        self._clean_formats(formats)
+        for stream in streams:
+            if isinstance(stream, str):
+                continue
+            if stream['type'] == 'video/mp4':
+                bitrate = self._search_regex(r'video-rrtv-(\d+)', stream['src'], 'vbr', None)
+                formats.append({
+                    'format_id': f'mp4-{bitrate}' if bitrate else 'mp4',
+                    'url': stream['src'],
+                    'ext': 'mp4',
```

Comment:
avoid using just `mp4` or `mp3` as the format_id since it conflicts with special arguments (`mp4` and `mp3` work like `best` or `bestaudio` etc)
```suggestion
                    'format_id': join_nonempty('http', ext, bitrate),
                    'url': stream['src'],
                    'ext': ext,
```